app/models/geometric-property.js

```diff
@@ -50,6 +51,51 @@ export const GEOMETRY_TYPES = [
   'rezoningArea',
 ];
 
+// returns true or false based on whether the change to the geometric
+// property was "meaningful"
+// first argument is a geometric-property model
+// todo: I think there are two separate handling here, one for
+// canonical geometries, one for regular
+export function isMeaningfulChange(geometricPropertyForType /* model */) {
+  // here, it gets set once by the constructor
+  // const initial = model.get(attribute);
+  const [
+    initial,
+    proposed, // upstream proposed should always be FC
+  ] = geometricPropertyForType.changedAttributes().proposedGeometry || [];
+
+  // "initial" means whatever is currently proposed. If it's _not_ empty and
+  // the geometric property doesn't have something canonical to compare,
+  // it should return true
+  if (!isEmpty(initial) && !geometricPropertyForType.get('hasCanonical')) return true;
+
+  // only apply this check if this is a canonical geometric prop
+  if (geometricPropertyForType.get('hasCanonical')) {
+    // check that proposed is not the canonical zoning
+    if ((!initial || isEmpty(initial)) && proposed) {
+      return isFeatureCollectionChanged(geometricPropertyForType.get('canonical'), proposed);
+    }
+
+    // if there is already initial, check that it's non-canonical
+    if (!isEmpty(geometricPropertyForType.get('proposedGeometry'))
+      && geometricPropertyForType.get('proposedDiffersFromCanonical')) {
+      return true;
+    }
+  }
+
+  // if nothing has been proposed at all, no
+  // meaningful changes detected
+  if (!proposed) return false;
+
+  // check for FC-ish empties
+  if (isEmpty(initial) && !isEmpty(proposed)) return true;
+
+  // finally, if the proposed is not empty, and it's a meaningful
+  // change in the feature collection, proceed
```

Comment:
`s/proceed/return true/` ...  we know that this boolean is used to control progression in the wizard, but in the context of this function, all that is happening is a true/false flag indicating whether meaningful changes were made. proceed is confusing out of context.

Reply:
@julialucyhogan it's very confusing. I think this comes down to the function using `return` to escape the sequence in multiple places. It should be rewritten.